src/page-container/usePageManager.jsx

```diff
@@ -0,0 +1,200 @@
+ * @param {Object} rootContainerRef A React ref object with a current value pointing
+ * to a valid HTMLElement
+ * @returns An object containing data related to currently active/visible Page
+ * and a unique PageManager component definition
+ */
+const usePageManager = (rootContainerRef) => {
+  const pageRegisterRef = React.useRef({});
+  const lastActivePageArrayRef = React.useRef([]);
+  const [activePageArray, setActivePageArray] = React.useState([]);
+
+  const pageManagerContextValue = React.useMemo(() => {
+    const registerPage = ({
+      portalElement,
+      pageId,
+      parentPageId,
+      label,
+      metaData,
+    }) => {
+      pageRegisterRef.current[pageId] = {
+        pageId,
+        portalElement,
+        parentPageId,
+        label,
+        metaData,
+        overflowData: pageRegisterRef.current[pageId]
+          ? pageRegisterRef.current[pageId].overflowData : undefined,
+      };
+
+      setActivePageArray((activePages) => {
+        // If the specified Page is already present in the render order,
+        // it can stay where it is. The state is updated to ensure that any changed
+        // page values (label/metaData) are communicated to the other effects.
+        if (activePages.indexOf(pageId) >= 0) {
+          return [...activePages];
+        }
+
+        // If there are currently no registered Pages, we initialize the state
+        // with the single new entry.
+        if (!activePages.length) {
+          return [pageId];
+        }
+
+        // If no ancestor Page is specified, but there are known active Pages,
+        // this new Page must be a duplicate at the root of the PageContainer.
+        // This new Page is not queued for rendering and is ignored.
+        if (!parentPageId) {
+          // We need to perform a separate check to see if the new Page should
+          // be inserted into the existing ordering, likely due to a series of
+          // Pages mounting at the same time.
+          const existingChild = pageRegisterRef.current[activePages[0]];
+          if (existingChild && existingChild.parentPageId === pageId) {
+            const newPages = [...activePages];
+            newPages.splice(0, 0, pageId);
+            return newPages;
+          }
+
+          Logger.warn(`[terra-application] A PageContainer cannot render multiple Pages with a shared parent. The redundant Page ${label} will not be displayed.`);
+          return activePages;
+        }
+
+        // If the new Page's parent has already been registered, the new
+        // Page is injected into the array immediately after the parent. If
+        // the parent is found to already have descendant Pages, the new Page
+        // is not queued for rendering and is ignored.
+        const indexOfParentPage = activePages.indexOf(parentPageId);
+        if (indexOfParentPage >= 0) {
+          if (indexOfParentPage === activePages.length - 1) {
+            const newPages = [...activePages];
```

Comment:
nitpick - same thinking here, except reversed. but I think what you have pretty straightforward too
```suggestion
            const newPages = [...activePages, pageId];
```